src/dataframe/dataframe.jl

```diff
@@ -43,6 +43,8 @@ A `DataFrame` is a lightweight object. As long as columns are not
 manipulated, creation of a DataFrame from existing AbstractVectors is
 inexpensive. For example, indexing on columns is inexpensive, but
 indexing by rows is expensive because copies are made of each column.
+An exception is assignment of `AbstractRange` as a column of
```

Comment:
I don't think "exception" is correct, as the previous sentence does not refer to columns passed to the constructor. Better add a separate paragraph, which would also document that column vectors are not copied in general.

I guess `setindex!` should be changed too, and the docstring be adapted?

Reply:
Second try to clarify things.

`setindex!` was changed already (but no `setindex!` methods needed changing directly). It does not have its docstring so I have left it as is for now (hopefully in the near future I will have time to go back to #1380 as it is a major redesign and then I will write an updated docstring).

Reply:
Has it? I see this:
```julia
julia> x = DataFrame(x=1:2)
2×1 DataFrames.DataFrame
│ Row │ x │
├─────┼───┤
│ 1   │ 1 │
│ 2   │ 2 │

julia> x[:x] = 2:3
2:3

julia> x[:x]
2:3

```

Reply:
Try testing with this PR checked out. You will see:
```
julia> x[:x] = 2:3
2:3

julia> x[:x]
2-element Array{Int64,1}:
 2
 3
```

Reply:
OK, that's what I realized after posting. But then it should be tested, right?

Reply:
This is exactly what lines 624 and 625 test.

Reply:
Damn, let's say it was too late in the evening... :-/